src/modules/test/main.cpp

```diff
@@ -0,0 +1,133 @@
+
+    if (json_value_get_type(root_value) != JSONArray)
+    {
+        json_value_free(root_value);
+        return testRecipes;
+    }
+
+    JSON_Array *jsonTestRecipesMetadata = json_value_get_array(root_value);
+    JSON_Object *jsonTestRecipeMetadata = nullptr;
+    for (size_t i = 0; i < json_array_get_count(jsonTestRecipesMetadata); i++)
+    {
+        jsonTestRecipeMetadata = json_array_get_object(jsonTestRecipesMetadata, i);
+        TestRecipeMetadata recipeMetadata = {
+            json_object_get_string(jsonTestRecipeMetadata, g_modulePath.c_str()),
+            json_object_get_string(jsonTestRecipeMetadata, g_mimPath.c_str()),
+            json_object_get_string(jsonTestRecipeMetadata, g_testRecipesPath.c_str()),
+        };
+
+        TestRecipes recipes = TestRecipeParser::ParseTestRecipe(recipeMetadata.m_testRecipesPath);
+        for (auto &recipe : *recipes)
+        {
+            recipe.m_metadata = recipeMetadata;
+            recipe.m_mimObjects = MimParser::ParseMim(recipeMetadata.m_mimPath);
+        }
+
+        testRecipes->insert(testRecipes->end(), recipes->begin(), recipes->end());
+    }
+
+    json_value_free(root_value);
+    return testRecipes;
+}
+
+TestRecipes LoadValuesFromCLI(char* modulePath, char* mimPath, char* testRecipesPath)
+{
+    TestRecipes testRecipes = std::make_shared<std::vector<TestRecipe>>();
+
+    TestRecipeMetadata recipeMetadata = {
+        modulePath,
+        mimPath,
+        testRecipesPath,
+    };
+
+    TestRecipes recipes = TestRecipeParser::ParseTestRecipe(recipeMetadata.m_testRecipesPath);
+    for (auto &recipe : *recipes)
+    {
+        recipe.m_metadata = recipeMetadata;
+        recipe.m_mimObjects = MimParser::ParseMim(recipeMetadata.m_mimPath);
+    }
+
+    testRecipes->insert(testRecipes->end(), recipes->begin(), recipes->end());
+
+    return testRecipes;
+}
+
+int main(int argc, char **argv)
+{
+    SetFullLogging(true);
+    testing::InitGoogleTest(&argc, argv);
+    TestRecipes testRecipes;
+    
+    if (argc == 2)
+    {
+        // Test definitions present - perform more extensive tests using Mim and test recipes
+        if (strcmp(".json", std::filesystem::path(argv[1]).extension().c_str()) == 0)
+        {
+            testRecipes = LoadValuesFromConfig();
+        }
+        else
+        {
+            // Perform basic module tests - MMI only
+            std::cout << "Performing basic module tests on '" << argv[1] << "'" << std::endl;
+            auto module = std::make_shared<ManagementModule>(argv[1]);
+
+            // See gtest.h for details on this test registration
+            // https://github.com/google/googletest/blob/v1.10.x/googletest/include/gtest/gtest.h#L2438
+            testing::RegisterTest(
+                "TestRecipes", "Basic-Module-Test", nullptr, "Basic-Module-Test", __FILE__, __LINE__,
+                    [module]() -> RecipeFixture* { return new BasicModuleTester(module); });
```

Comment:
>[module]() -> RecipeFixture* { return new BasicModuleTester(module); }); [](http://example.com/codeflow?start=20&length=72)

Break this line and re-arrange as following:

    [module]()->RecipeFixture* 
    { 
        return new BasicModuleTester(module); 
    }
);